In [66]:
import pandas as pd
import matplotlib.pyplot as plt

# importando os dois datasets para a classe DataFrame do Pandas

df_quiz = pd.read_csv('./case estágio de dados/quiz-sample.csv')
df_orders = pd.read_csv('./case estágio de dados/orders-sample.csv')

df_orders['cancelled_at'].isnull().value_counts()

True     25485
False     4746
Name: cancelled_at, dtype: int64

In [57]:
# criando um novo DataFrame com os dados da person_id relacionados de uma tabela na outra

df_merge = pd.merge(df_quiz, df_orders, on=['person_id'], how='inner').drop_duplicates(['person_id', 'preocupacao_de_pele']).reset_index()

# merge -> faz a unificação dos dados que possuem relação, neste caso, person_id dos dois DataFrames
# drop_duplicates -> para remover pessoas duplicadas do DataFrame (existiam pessoas que tinha uma mesma preocupação, então retirei
#                    as que possuiam essas duplicações)
# reset_index -> para colocar os index em ordem

df_merge

index                                     person_id  genero  \
0            0  6bobH7lLMcVLVmXd5RYZWQCNtw/NNfb6CG1nvT/1pfc=  Mulher   
1            1  6bobH7lLMcVLVmXd5RYZWQCNtw/NNfb6CG1nvT/1pfc=  Mulher   
2            2  6bobH7lLMcVLVmXd5RYZWQCNtw/NNfb6CG1nvT/1pfc=  Mulher   
3            3  6bobH7lLMcVLVmXd5RYZWQCNtw/NNfb6CG1nvT/1pfc=  Mulher   
4            4  6bobH7lLMcVLVmXd5RYZWQCNtw/NNfb6CG1nvT/1pfc=  Mulher   
...        ...                                           ...     ...   
132739  210950  RAjxQiRQNjiAJe9P0xtUU/B2oil4sgYCFDKicJCGqS0=  Mulher   
132740  210952  RAjxQiRQNjiAJe9P0xtUU/B2oil4sgYCFDKicJCGqS0=  Mulher   
132741  210954  RAjxQiRQNjiAJe9P0xtUU/B2oil4sgYCFDKicJCGqS0=  Mulher   
132742  210956  RAjxQiRQNjiAJe9P0xtUU/B2oil4sgYCFDKicJCGqS0=  Mulher   
132743  210958  RAjxQiRQNjiAJe9P0xtUU/B2oil4sgYCFDKicJCGqS0=  Mulher   

                     idade tipo_pele  pele_sensivel            estado_vive  \
0       Entre 25 e 29 anos   Não sei              0  Distrito Federal - DF   
1       Entre 25 e 29 anos   Não sei              0  Distrito Federal - DF   
2       Entre 25 e 29 anos   Não sei              0  Distrito Federal - DF   
3       Entre 25 e 29 anos   Não sei              0  Distrito Federal - DF   
4       Entre 25 e 29 anos   Não sei              0  Distrito Federal - DF   
...                    ...       ...            ...                    ...   
132739  Entre 35 e 39 anos     Mista              1         São Paulo - SP   
132740  Entre 35 e 39 anos     Mista              1         São Paulo - SP   
132741  Entre 35 e 39 anos     Mista              1         São Paulo - SP   
132742  Entre 35 e 39 anos     Mista              1         São Paulo - SP   
132743  Entre 35 e 39 anos     Mista              1         São Paulo - SP   

       data_submissao               preocupacao_de_pele  tem_preocupacao  \
0          2021-06-22            pp_bolsas_abaixo_olhos                0   
1          2021-06-22             pp_cicatrizes_espinha                0   
2          2021-06-22  pp_cicatrizes_machucado_cirurgia                0   
3          2021-06-22                         pp_cravos                1   
4          2021-06-22                     pp_descamacao                0   
...               ...                               ...              ...   
132739     2019-09-02                          pp_rugas                0   
132740     2019-09-02                         pp_sardas                0   
132741     2019-09-02             pp_vasinhos_aparentes                0   
132742     2019-09-02                    pp_vermelhidao                0   
132743     2019-09-02         pp_nao_tenho_preocupacoes                0   

        product_A  product_B  product_C  product_D  product_E  \
0               0          0          1          0          0   
1               0          0          1          0          0   
2               0          0          1          0          0   
3               0          0          1          0          0   
4               0          0          1          0          0   
...           ...        ...        ...        ...        ...   
132739          0          1          0          1          0   
132740          0          1          0          1          0   
132741          0          1          0          1          0   
132742          0          1          0          1          0   
132743          0          1          0          1          0   

                   processed_at cancelled_at financial_status    billing_city  \
0       2020-04-26 11:10:32 UTC          NaN             paid  Rio de Janeiro   
1       2020-04-26 11:10:32 UTC          NaN             paid  Rio de Janeiro   
2       2020-04-26 11:10:32 UTC          NaN             paid  Rio de Janeiro   
3       2020-04-26 11:10:32 UTC          NaN             paid  Rio de Janeiro   
4       2020-04-26 11:10:32 UTC          NaN             paid  Rio de Janeiro   
...                        

In [54]:
# ver quantas preocupações existiam no DataFrame original

preocupacoes = df_quiz[['preocupacao_de_pele']].drop_duplicates().sort_values('preocupacao_de_pele').set_index('preocupacao_de_pele')

preocupacoes

Empty DataFrame
Columns: []
Index: [pp_bolsas_abaixo_olhos, pp_cicatrizes_espinha, pp_cicatrizes_machucado_cirurgia, pp_cravos, pp_descamacao, pp_excesso_brilho, pp_flacidez_rosto, pp_linhas_finas, pp_manchas_brancas, pp_manchas_escuras, pp_melasma, pp_muitas_espinhas, pp_nao_tenho_preocupacoes, pp_oleosidade, pp_olheiras, pp_pintas, pp_poros_dilatados, pp_poucas_espinhas, pp_ressecamento, pp_rosacea, pp_rugas, pp_sardas, pp_vasinhos_aparentes, pp_vermelhidao]

In [62]:
# contar quantas preocupações existem em cada uma das suas classes

# df_merge.groupby(['preocupacao_de_pele']).count()

quantidade = df_merge['preocupacao_de_pele'].value_counts()

quantidade

pp_bolsas_abaixo_olhos              5531
pp_cicatrizes_espinha               5531
pp_vermelhidao                      5531
pp_vasinhos_aparentes               5531
pp_sardas                           5531
pp_rugas                            5531
pp_rosacea                          5531
pp_ressecamento                     5531
pp_poros_dilatados                  5531
pp_pintas                           5531
pp_olheiras                         5531
pp_oleosidade                       5531
pp_melasma                          5531
pp_manchas_escuras                  5531
pp_manchas_brancas                  5531
pp_linhas_finas                     5531
pp_flacidez_rosto                   5531
pp_excesso_brilho                   5531
pp_poucas_espinhas                  5531
pp_muitas_espinhas                  5531
pp_descamacao                       5531
pp_cravos                           5531
pp_cicatrizes_machucado_cirurgia    5531
pp_nao_tenho_preocupacoes           5531
Name: preocupaca

In [63]:
# verificação por "força bruta", para verificar se o resultado bate com o de cima

cont = 0

for i in range(len(df_merge)):
    try:
        if(df_merge['preocupacao_de_pele'][i] == 'pp_oleosidade'):
            cont += 1
    except:
        pass

cont

5531

## B) Considerando as bases de quiz e compras, responda:

### 1. Quais as 5 preocupações de pele mais recorrentes entre os nossos clientes a cada mês, considerando apenas respostas de pessoas que possuam pedidos registrados na base de compras?

Observando os dados e fazendo a análise supracitada, cheguei na informação de que todas as preocupações equiparam-se, sendo o mesmo valor para todas elas. Todas as preocupações obtiveram 5531 submissões, onde essas pessoas que fizeram a submissão também possuem compras registradas na base de compras. Desta forma, todas elas possuem o mesmo valor de submissões e são iguais.